src/components/DatePicker/CalendarDialog.tsx

```diff
@@ -0,0 +1,169 @@
+`
+
+interface CalendarDialogProps {
+  date: Date | null
+  title?: string
+  timeFrame?: TimeFrame
+  monthLabel: string
+  yearLabel: string
+  info?: string
+  isOpen: boolean
+  locale: string
+  close: () => void
+  onChange: (date: Date) => void
+}
+
+const CalendarDialog = ({
+  date,
+  title,
+  timeFrame,
+  monthLabel,
+  yearLabel,
+  info,
+  isOpen,
+  close,
+  locale,
+  onChange,
+}: CalendarDialogProps) => {
+  const month = date?.getMonth() || new Date().getMonth()
+  const year = date?.getFullYear() || new Date().getFullYear()
+
+  const [selectedMonth, setSelectedMonth] = useState(month)
+  const [selectedYear, setSelectedYear] = useState(year)
+
+  const daysInMonth = new Date(selectedYear, selectedMonth + 1, 0).getDate()
+  const firstDayOfMonth = new Date(selectedYear, selectedMonth, 1).getDay()
+
+  const weekdays = getWeekdays(locale)
+  const months = getMonths(locale)
+  const years = getYears(timeFrame)
+
+  const days: Array<React.ReactNode> = []
+
+  for (let i = 1; i < daysInMonth + 1; i++) {
+    const day = new Date(selectedYear, selectedMonth, i)
+    const isDisabled =
+      (timeFrame === TimeFrame.future && isPastDate(day)) ||
+      (timeFrame === TimeFrame.past && isFutureDate(day))
+
+    days.push(
+      <Day
+        key={i}
+        date={day}
+        isDisabled={isDisabled}
+        isSelected={!!date && isSameDate(date, day)}
+        onSelect={onChange}
+      />
+    )
+  }
+
+  for (let i = 1; i < firstDayOfMonth; i++) {
+    days.unshift(<div key={`prev-${i}`} />)
+  }
+
+  return (
+    <Dialog on={isOpen}>
+      {({ getWindowProps }) => (
+        <DialogWindow
+          {...getWindowProps({
+            onClick: close,
+            onEscKeyDown: close,
+          })}
+        >
+          <DialogHeader>
+            {title}
+            <DialogAdornment right>
+              <button onClick={close}>
+                <CloseAlt size={24} />
+              </button>
+            </DialogAdornment>
+          </DialogHeader>
+          <DialogBody>
+            <Selects>
+              <Select
+                id="month"
+                items={months}
+                label={monthLabel}
+                initialSelectedItem={months.find(
+                  i => i.value === String(month)
+                )}
+                hideLabel
+                onChange={selected => setSelectedMonth(Number(selected.value))}
+              />
+              <Select
+                id="year"
+                items={years}
+                label={yearLabel}
+                initialSelectedItem={years.find(i => i.value === String(year))}
+                hideLabel
+                onChange={selected => setSelectedYear(Number(selected.value))}
+              />
+            </Selects>
+            <CalendarTitles>
+              {weekdays.map((weekday, index) => (
+                <CalendarTitle key={`day-${index}`}>{weekday}</CalendarTitle>
+              ))}
+            </CalendarTitles>
```

Comment:
Not sure if this is because Storybook's viewport changer because I only have this if I use there option to change to viewport to 'small mobile' but when I do it looks like this:

![image](https://user-images.githubusercontent.com/12168698/145973404-3fb20c51-1e0c-4c5c-8e62-bc6a1c9ebdd1.png)

Reply:
Nice catch! This was not okay on mobile. Should be fixed now, let me know what you think. 🙂 

<img width="596" alt="Screenshot 2021-12-14 at 16 29 20" src="https://user-images.githubusercontent.com/1096800/146029664-f8ce4cda-109d-4bb9-9871-91d266f4b777.png">

<img width="318" alt="Screenshot 2021-12-14 at 16 28 58" src="https://user-images.githubusercontent.com/1096800/146029657-b1dc3f8b-f874-4fc3-88f2-c9ad18b6b2c2.png">

Reply:
Still happens with me

Reply:
This is really weird, maybe we can make a call to check this? I tested on my phone and all mobile phones on Chrome Dev Tools and seems okay.

https://user-images.githubusercontent.com/1096800/146166741-6818dc3e-c51c-4d68-8ccb-99c7dc264461.MP4